src/bindings/python/src/openvino/frontend/tensorflow/utils.py

```diff
@@ -0,0 +1,204 @@
+    import tensorflow as tf
+    if not isinstance(input_model, (tf.keras.layers.Layer, tf.Module, tf.keras.Model, tf.types.experimental.GenericFunction)):
+        return input_model
+    return trace_tf_model(input_model, placeholder_shapes, placeholder_data_types, example_input)
+
+
+def get_input_spec_from_model(model):
+    import tensorflow as tf
+    if hasattr(model, "_build_input_shape") and model._build_input_shape is not None:
+        if isinstance(model._build_input_shape, list):
+            input_spec = [[tf.TensorSpec(shape) for shape in model._build_input_shape]]
+        else:
+            input_spec = [tf.TensorSpec(model._build_input_shape)]
+    else:
+        input_spec = [tf.TensorSpec(None)]
+    return input_spec
+
+
+def create_example_input_by_user_shapes(input_shapes, input_types):
+    import tensorflow as tf
+    if input_shapes is None:
+        return None
+    if isinstance(input_shapes, dict):
+        res = {}
+        for name, shape in input_shapes.items():
+            shape = get_static_shape(shape, 1)
+            args = {}
+            if name in input_types:
+                args["dtype"] = input_types[name]
+            tensor = tf.random.uniform(shape=shape, **args)
+            res[name] = tensor
+        return res
+    elif isinstance(input_shapes, list):
+        res = []
+        for idx, shape in enumerate(input_shapes):
+            shape = get_static_shape(shape, 1)
+            args = {}
+            if idx < len(input_types):
+                args["dtype"] = input_types[idx]
+            tensor = tf.random.uniform(shape=shape, **args)
+            res.append(tensor)
+        return res
+    raise Error("Could not create example input by provided shape {}".format(input_shapes))
+
+
+def get_concrete_func(tf_function, example_input, input_needs_packing, error_message, use_example_input=True):
+    if input_needs_packing and not isinstance(example_input, (list, tuple)):
+        example_input = [example_input]
+    try:
+        if use_example_input:
+            if not input_needs_packing and isinstance(example_input, (list, tuple)):
+                concrete_func = tf_function.get_concrete_function(*example_input)
+            else:
+                concrete_func = tf_function.get_concrete_function(example_input)
+
+        else:
+            concrete_func = tf_function.get_concrete_function()
+    except Exception as e:
+        raise Exception(error_message.format(e))
+    return concrete_func
+
+
+def trace_tf_model(model, input_shapes, input_types, example_input):
+    import tensorflow as tf
+    if isinstance(model.__call__, tf.types.experimental.GenericFunction):
+        tf_function = model.__call__
+        input_needs_packing = False
+    elif isinstance(model, tf.types.experimental.GenericFunction):
+        tf_function = model
+        input_needs_packing = False
+    else:
+        # Wrap model to tf.Function
+        @tf.function
+        def tf_function(args):
```

Comment:
in this case we don't have (input tensor names) names of inputs. Is it correct? May be add a comment. Can be addressed in the next PR.

Reply:
Added comment.